### PR TITLE
Release 0.10.0

### LINKED ISSUE #485: Release cycle
@danilobellini what do you think to publish releases more often? For instance there are several bug fixes in the current unreleased version. Perhaps could you publish a 0.9.1 with those changes?

### PR DESCRIPTION
**Pull request by BoboTiG**
The so-waited new release! Will close #485.

@danilobellini I will merge only with your approval, I am not quite sure if this is all to mark that new release. Also, when merged, would you mind upload the new version to PyPi?

### CODE REVIEW

**Comment by BoboTiG** [comment]
If was thinking of making the release by the end of the year, when you have 5 minutes free :crossed_fingers: :)

**Comment by hazmat345** [comment]
Hi @BoboTiG, @danilobellini - I'm not seeing this release (0.10.0) on pypi.org, was it published there? Apologies if I'm missing something, just want to make sure that didn't fall through the cracks.

Thank you both for all your work on this!

**Comment by Ajordat** [comment]
I'm having the same issue as @hazmat345, I don't see v0.10.0 on PyPi and I'd love to install the latest version on my project!

**Comment by BoboTiG** [comment]
Sorry for the delay. I am trying to reach out the project author to be able to make the release on PyPi.
I hope it will be possible by the next week.

**Comment by danilobellini** [comment]
Sorry people, my computer broke and I was (am!) in a hurry to finish some of my work that delayed a lot because of that, and I had no time yet to see/review this.

I'm not the project owner/author but I can easily push it to PyPI from the tag version. Is it okay for you that I'm gonna push a code there that I haven't reviewed yet? @BoboTiG, if you're confident enough to "pull the trigger", I'll upload it there today.

There's only a single issue I can remind that we should at least be aware of, it's about the readthedocs: I'm the owner of https://python-watchdog.readthedocs.io/en/latest/ but I don't know who is the owner of https://watchdog.readthedocs.io/en/latest/ (I asked @gorakhargosh and he said he's not the owner of it, either).

**Comment by BoboTiG** [comment]
@danilobellini I am confident to upload the release now. Just checkout the [https://github.com/gorakhargosh/watchdog/tree/v0.10.0](v0.10.0) tag and it is good to upload as-is. It was what I was trying to do before I hit the HTTP 403 error :)

Is it an issue the readhedocs thing? As it is up-to-date with master, maybe this is good as-is for now?
Anyway, the tag is already created and I think we should just upload the release without more work (and see how we broke the entire world haha).

**Comment by danilobellini** [comment]
`HTTPError: 400 Client Error`: The description failed to render in the default format of reStructuredText. See https://pypi.org/help/#description-content-type for more information.

I can fix it, that's because of the `====` with a `~~~~` afterwards in the changelog file (GitHub doesn't complain about it, but http://rst.ninjs.org does, and that's probably why PyPI is complaining now). Replacing it by `**` should fix it.

I'll create a new commit in a new `v0.10.x` branch with these small changes that are going to be uploaded to PyPI. I think we should move the `v0.10.0` tag as well to match the version in PyPI.

**Comment by BoboTiG** [comment]
OK, let's do that. :+1:

**Comment by danilobellini** [comment]
It's done! And https://python-watchdog.readthedocs.io/en/latest/ was updated, as well.

I recommend merging the v0.10.x branch into master, updating the v0.10.0 tag to the new commit, and perhaps updating its release date in the changelog to today (2020-01-26), so it'll match the new commit.

It's strange that the tag isn't an annotated one in git, but it's annotated in GitHub.

**Comment by BoboTiG** [comment]
After 2 years, finally :) Thanks!

Can you manage the tag update?

**Comment by BoboTiG** [comment]
There is a commit between the old tag and your commit. History will be hard to follow, no?

**Comment by danilobellini** [comment]
> There is a commit between the old tag and your commit. History will be hard to follow, no?

Don't worry, it's a simple merge, and one that makes historical sense. The code is no longer in a "guitar hero"-like merge hell. =)

I already merged and removed the `v0.10.x` branch (though it can be created again, if required).

> After 2 years, finally :) Thanks!

Huahauhauaha yes! I think we can release more often.

> Can you manage the tag update?

Done! From `git` it was simple (`git tag -f v0.10.0` and a push), but only v0.6.0 is an annotated tag. The release information is stored by GitHub, not git, and I found no way to update the release date as shown by GitHub, so I removed the release and released it again. =)

Perhaps we should consider moving the release information in GitHub to annotated tags. The annotated tags are the ones intended for releases (the other ones are just a reference to a commit, e.g. for CI automation).

**Comment by BoboTiG** [comment]
> Perhaps we should consider moving the release information in GitHub to annotated tags. The annotated tags are the ones intended for releases (the other ones are just a reference to a commit, e.g. for CI automation).

Of course!

**Comment by Ajordat** [comment]
Thank you very much, guys!

**Comment by hazmat345** [comment]
Thanks!!!
